### ISSUE TITLE
`@ApplicationModuleTest` unable to bootstrap module

### ISSUE THREAD

**Issue by kmartin-bh** [spring-projects/spring-modulith#449]
I'm having trouble creating integration tests for my modules. This is occurring for all modules, whether or not the module has any direct dependencies.
spring-modulith version `1.1.1`.

My Test
```
package com.companyname.appname.organization;

import lombok.AllArgsConstructor;
import org.junit.jupiter.api.Test;
import org.springframework.modulith.test.ApplicationModuleTest;

@ApplicationModuleTest
@AllArgsConstructor
public class OrganizationIntegrationTest {
  @Test
  void bootstrapsOrganizationModule() {

  }
}
```

Test Output
```
2024-01-10 15:16:27.042 INFO [ | c.t.a.c.PluginLoader$PluginSupplier:67] Detected Java version 17.0.2  
2024-01-10 15:16:33.091 INFO [ | o.s.m.t.ModuleContextCustomizerFactory$ModuleContextCustomizer:101] Bootstrapping @org.springframework.modulith.test.ApplicationModuleTest for Organization in mode STANDALONE (class com.bamboohealth.jimgmtapp.JiMgmtApplication)…  
2024-01-10 15:16:33.092 INFO [ | o.s.m.t.ModuleContextCustomizerFactory$ModuleContextCustomizer:102]   
2024-01-10 15:16:33.104 INFO [ | j.u.Spliterators$ArraySpliterator:992] # Organization  
2024-01-10 15:16:33.104 INFO [ | j.u.Spliterators$ArraySpliterator:992] > Logical name: organization  
2024-01-10 15:16:33.104 INFO [ | j.u.Spliterators$ArraySpliterator:992] > Base package: com.bamboohealth.jimgmtapp.organization  
2024-01-10 15:16:33.105 INFO [ | j.u.Spliterators$ArraySpliterator:992] > Direct module dependencies: none  
2024-01-10 15:16:33.105 INFO [ | j.u.Spliterators$ArraySpliterator:992] > Spring beans:  
2024-01-10 15:16:33.105 INFO [ | j.u.Spliterators$ArraySpliterator:992]   + ….OrganizationService  
2024-01-10 15:16:33.105 INFO [ | j.u.Spliterators$ArraySpliterator:992]   o ….internal.OrgDynamoDbTable  
2024-01-10 15:16:33.106 INFO [ | o.s.m.t.ModuleContextCustomizerFactory$ModuleContextCustomizer:143]   
2024-01-10 15:16:33.120 INFO [ | o.s.b.StartupInfoLogger:55] Starting OrganizationIntegrationTest using Java 17.0.2 on kmartin-rmac-1121 with PID 244 (started by kmartin in /Users/kmartin/Appriss/projects/JI/ji-mgmt)  
2024-01-10 15:16:33.120 INFO [ | o.s.b.StartupInfoLogger:55] Starting OrganizationIntegrationTest using Java 17.0.2 on kmartin-rmac-1121 with PID 244 (started by kmartin in /Users/kmartin/Appriss/projects/JI/ji-mgmt)  
2024-01-10 15:16:33.121 INFO [ | o.s.b.SpringApplication:640] The following 1 profile is active: "test"  
2024-01-10 15:16:33.121 INFO [ | o.s.b.SpringApplication:640] The following 1 profile is active: "test"  
2024-01-10 15:16:33.907 INFO [ | o.s.m.t.ModuleTestAutoConfiguration$AutoConfigurationAndEntityScanPackageCustomizer:63] Re-configuring auto-configuration and entity scan packages to: com.bamboohealth.jimgmtapp.organization.  
2024-01-10 15:16:33.938 ERROR [ | o.s.b.SpringApplication:824] Application run failed java.lang.NullPointerException: Cannot invoke "org.springframework.beans.factory.config.ConstructorArgumentValues$ValueHolder.getValue()" because "holder" is null ¶ 	at org.springframework.modulith.test.ModuleTestAutoConfiguration$AutoConfigurationAndEntityScanPackageCustomizer.setBasePackagesOn(ModuleTestAutoConfiguration.java:80) ¶  
2024-01-10 15:16:33.942 ERROR [ | o.s.t.c.TestContextManager:252] Caught exception while allowing TestExecutionListener [org.springframework.test.context.web.ServletTestExecutionListener@4f9a2c08] to prepare test instance [com.bamboohealth.jimgmtapp.organization.OrganizationIntegrationTest@10754d5c] java.lang.NullPointerException: Cannot invoke "org.springframework.beans.factory.config.ConstructorArgumentValues$ValueHolder.getValue()" because "holder" is null ¶ 	at org.springframework.modulith.test.ModuleTestAutoConfiguration$AutoConfigurationAndEntityScanPackageCustomizer.setBasePackagesOn(ModuleTestAutoConfiguration.java:80) ¶ Wrapped by: java.lang.IllegalStateException: Failed to load ApplicationContext ¶ 	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:132) ¶  

Process finished with exit code 255
```

My `modules.verify()` test is passing. Would this error suggest that I have some issues in my module's dependency injection? Or is this a bug with the `org.springframework.modulith.test.ModuleTestAutoConfiguration` class? 

I'm dealing with legacy code, so getting an existing application to conform to the DDD standards is a process.

**Comment by odrotbohm** [comment]
So the error is triggered by either the bean definition for `AutoConfigurationPackages` or `EntityScanPackages` not getting a constructor argument set, which, from looking at the way those get registered, I can't see a way in which this would happen at all. I can easily add a guard for that but in case you have a simple reproducer project at hand, I'd love to investigate why this happens even further.

**Comment by kmartin-bh** [comment]
@odrotbohm Thanks for the reply - I also was unable to figure out why this would happen. Debugging through the code didn't give me much. At the moment I only have my enterprise application as an example, and I'm still working through implementing spring-modulith.

Once everything is ready to go, I will try to replicate this bug in an example application that I can share.

**Comment by odrotbohm** [comment]
This is now fixed and back ported to 1.1.x and 1.0.x.

**Comment by kmartin-bh** [comment]
Thanks for the quick turnaround! I'll test it on my end as well.
